project_overview.md

```diff
@@ -0,0 +1,64 @@
+* __FullWorkArea__: Used by the `CageDesktop` class to initialize and reserve space for the `CageLabeler`.`
+* __SecuritySetup__: Sets up and populates a [SECURITY_ATTRIBUTES](https://msdn.microsoft.com/en-us/library/windows/desktop/aa379560%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396) struct used for example to create the secure desktop.
+
+## CageService
+
+* __CageService__: Checks if configuration files gotten as messages from `CageChooser` (and passed on by `CageServiceMain`) have the right access rights, asserts if the `CageManager` binary has the correct certificate and then starts it in the context of the current user (passing on the configuration file as well). Afterwards it waits until the `CageManager` instance closes again and only then accepts further messages (ensuring only one secure desktop can run at the same time).
+* __CageServiceMain__: Starts the `CageService`, continuously waits for messages to arrive (& pass them to the `CageService` instance) and waits for shutdown event in case the windows service gets stopped.
+
+## SharedFunctionality
+
+Gets compiled as a `.lib` file and statically linked by the other sub-projects. For use in C# it also gets compiled to as `.dll` file and loaded dynamically by the `.NET Framework` (P/Invoke).
+
+### Network
+
+* __Messages__: Contains all messages currently in use by project.
+* __NetworkManager__: Handles network communication between the different binaries. When creating an instance a port used for listening must be specified, but outgoing messages can be addressed to all ports used by any of the other binaries / projects. After sending a message the `NetworkManager` briefly waits for a response and reports the result to the caller.
+
+### TokenLib
+
+* __groupManipulation__: Used for creating and deleting local user groups.
+* __tokenManipulation__: Used for manipulating windows access tokens.
+
+### Misc
+
+* __CageData__: Contains a struct which encapsulates all data used by various functions responsible for displaying data / creating processes in the secure desktop.
+* __SharedFunctions__: Compilation of helper functions used by most of the other sub-projects.
+* __ValidateBinary__: Functions for validating binary certificates (by using the appropriate Windows Crypto-API (`CNG`) functions) and hashes (by comparing them).
+
+## SharkCageInstallerHelper
+
+* __SharkCageInstallerHelper__: Loaded by the installer. Contains functions which further customize the installer behaviour with functionality not available by default.
+
+## SharkCageInstaller
+
+Installer project which is able to configure target systems for usage of Shark Cage (either user guided or silently).
+
+# Sequence diagram of a typical invocation of Shark Cage
+
+![](sharkcage_sequence_diagramm.png)
```

Comment:
Maybe we can replace this link with that one: `![sharkcage_sequence_diagramm](https://user-images.githubusercontent.com/1786772/44297848-e5646e00-a2d8-11e8-9fbc-9a2cf502e608.png)`

Then we can delete the image from the repo: https://github.com/SharkCagey/HTWG_shark_cage/blob/111-component-overview/sharkcage_sequence_diagramm.png
I think we should keep as few non-source files as possible directly in the repo.